UiPath.FreeRdpClient/UiPath.FreeRdpClient/Internals/FreeRdpClient.cs

```diff
@@ -63,10 +67,14 @@ public async Task<IAsyncDisposable> Connect(RdpConnectionSettings connectionSett
         Task<AsyncDisposable> DoConnect() => Task.Run(() =>
         {
             NativeInterface.RdpLogon(connectOptions, out var releaseObjectName);
-            return new AsyncDisposable(async () =>
+            var connection = new AsyncDisposable(async () =>
             {
                 Disconnect(releaseObjectName);
             });
+
+            _disconnectCallbacks.Add(releaseObjectName, connectionSettings.DisconnectCallback);
```

Comment:
releaseObjectname is based in scopename, maybe we tink if we prefer to throw here as it already exists or just override